akka-actor-typed-tests/src/test/scala/docs/akka/typed/RouterSpec.scala

```diff
@@ -143,5 +143,58 @@ class RouterSpec extends ScalaTestWithActorTestKit("akka.loglevel=warning") with
 
       probe.receiveMessages(10)
     }
+    "show group routing with consistent hashing" in {
+
+      val probe1 = createTestProbe[String]()
+      val probe2 = createTestProbe[String]()
+
+      object Proxy {
+
+        val RegisteringKey = ServiceKey[Command]("aggregator-key")
+
+        def mapping(command: Command) = command.id
+
+        sealed trait Command {
+          def id: String
+        }
+        case class Message(id: String, content: String) extends Command
+
+        def apply(monitor: ActorRef[String]): Behavior[Command] =
+          Behaviors.receiveMessage {
+            case Message(id, _) =>
+              monitor ! id
+              Behaviors.same
+          }
+      }
+
+      //registering proxies
+      val proxy1 = spawn(Proxy(probe1.ref))
+      val proxy2 = spawn(Proxy(probe2.ref))
+      val waiterProbe = createTestProbe[Receptionist.Registered]()
+
+      system.receptionist ! Receptionist.Register(Proxy.RegisteringKey, proxy1, waiterProbe.ref)
+      system.receptionist ! Receptionist.Register(Proxy.RegisteringKey, proxy2, waiterProbe.ref)
+      //wait until both registrations get Receptionist.Registered
+      waiterProbe.receiveMessages(2)
+
+      //messages sent to a router with constant hashing
+      // #constant-hashing
+      val router = spawn(Routers.group(Proxy.RegisteringKey).withConsistentHashingRouting(10, Proxy.mapping))
+
+      router ! Proxy.Message("123", "Text1")
+      router ! Proxy.Message("123", "Text2")
+
+      router ! Proxy.Message("zh3", "Text3")
+      router ! Proxy.Message("zh3", "Text4")
+      // the hash is calculated over the Proxy.Message first parameter obtained through the Proxy.mapping function
+      // #constant-hashing
+      //Then messages with equal Message.id reach the same actor
+      //so the first message in each probe queue is equal to its second
+      probe1.receiveMessage() shouldBe probe1.receiveMessage()
+      probe2.receiveMessage() shouldBe probe2.receiveMessage()
```

Comment:
Same here, this assertion can't really pass.

Reply:
I'm wrong, see my comment [here](https://github.com/akka/akka/pull/30036#discussion_r604936756).